kod/object/active/holder/room/necarena.kod

```diff
@@ -136,19 +141,6 @@ messages:
       return FALSE;
    }
 
-   CheckWatcher(what=$)
-   {
-      if poWatcher = $
-      {
-         debug("Goad is nil!  Goad should never be nil!  Goad was moved!");
-         poWatcher = what;
-         Send(self,@NewHold,#what=what,#new_row=13,#new_col=20,
-              #fine_row=0,#new_angle=ANGLE_WEST);
-      }
-
-      return;
-   }
```

Comment:
Check watcher was only ever called from Goad.  The problem is this check watcher implementation only looks at the poWatcher property which can easily be set to the Goad object while Goad is in another room.

This functionality (warping Goad back into NecroArena) was moved to Goad.

Reply:
Additionally - we can deduce that this function has never been called in production as the code that called this function was missing the what parameter.  The result of this is that the above code sends a new hold to $ - which I think would warp Goad into the void and leave him without an owner.